Makefile

```diff
@@ -1,13 +1,17 @@
-.PHONY: all test clean
+.PHONY: all installdeps build test clean
 
-all: .pre-build
-	@cargo make build
+all: .pre-build build
 
 .pre-build:
```

Comment:
Should this include installing the submodules too?

If so, we can update the ci workflows too

Reply:
Awesome, I meant that ci should use the make target (which includes installing submodules), instead of installing submodules on its own. Can be addressed in the future though :)